pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -293,6 +293,7 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 	c.podsClient = kclient.NewFiltered[*v1.Pod](kubeClient, kclient.Filter{
 		ObjectFilter:    kubeClient.ObjectFilter(),
 		ObjectTransform: kubelib.StripPodUnusedFields,
+		FieldSelector:   "status.phase!=Failed",
```

Comment:
with a field selector in k8s if it transitions to Failed do we get a delete event? I assume so

Reply:
I think so. If it is selected to start with, we should get delete event.